src/controllers/courses.controller.js

```diff
@@ -0,0 +1,92 @@
+const Course = require("../models/courses.model")
+const Step = require("../models/steps.model");
+
+// Validation Middleware
+
+// Checks if step exists
+function stepExists(req, res, next) {
+  const { stepId } = req.params;
+  Step.findById(stepId, (err, step) => {
+    if (err) return next();
+    if (!step) return next({
+      status: 404,
+      message: `Step id ${req.params.stepId} cannot be found`
+    });
+    res.locals.step = step;
+    next();
+  });
+}
+
+// Router-level Middleware
+
+// Create a Course
+function create(req, res, next) {
+  Course.create({ ...req.body }, (err, newCourse) => {
+    if (err) return next(err);
+    res.status(201).json({ message: "New Course Created", newCourse });
+  });
+}
+
+// Update a single course (only updates course 'title')
+function update(req, res, next) {
+  const { title } = req.body;
+  Course.findByIdAndUpdate(req.params.courseId, { title }, { new: true }, (err, course) => {
```

Comment:
You need a validator to check that the title is indeed sent, so the value does not get over-ridden by an empty request body

Reply:
Resolved.